src/main/java/sirius/biz/storage/layer3/VirtualFile.java

```diff
@@ -0,0 +1,794 @@
+        }
+    }
+
+    /**
+     * Tries to delete this file.
+     *
+     * @return <tt>true</tt> if the operation was successful, <tt>false</tt> otherwise
+     */
+    public boolean tryDelete() {
+        try {
+            if (deleteHandler == null) {
+                return false;
+            }
+
+            return deleteHandler.apply(this);
+        } catch (Exception e) {
+            throw handleErrorInCallback(e, "deleteHandler");
+        }
+    }
+
+    /**
+     * Deletes this file.
+     *
+     * @throws HandledException if the file cannot be deleted
+     */
+    public void delete() {
+        if (!tryDelete()) {
+            throw Exceptions.createHandled().withNLSKey("VirtualFile.cannotDelete").set("file", path()).handle();
+        }
+    }
+
+    /**
+     * Determines if this file can (probably) be renamed.
+     *
+     * @return <tt>true</tt> if this file can be renamed or <tt>false</tt> otherwise
+     */
+    public boolean canRename() {
+        try {
+            if (canRenameHandler != null) {
+                return canRenameHandler.test(this);
+            }
+
+            return renameHandler != null;
+        } catch (Exception e) {
+            throw handleErrorInCallback(e, "canRenameHandler");
+        }
+    }
+
+    /**
+     * Tries to rename this file.
+     *
+     * @return <tt>true</tt> if the operation was successful, <tt>false</tt> otherwise
+     */
+    public boolean tryRename(String newName) {
+        try {
+            if (renameHandler == null) {
+                return false;
+            }
+
+            return renameHandler.apply(this, newName);
+        } catch (Exception e) {
+            throw handleErrorInCallback(e, "renameHandler");
+        }
+    }
+
+    /**
+     * Renames this file.
+     *
+     * @throws HandledException if the file cannot be renamed
+     */
+    public void rename(String newName) {
+        if (!tryRename(newName)) {
+            throw Exceptions.createHandled().withNLSKey("VirtualFile.cannotRename").set("file", path()).handle();
+        }
+    }
+
+    /**
+     * Determines if this file can (probably) be move in an efficient way.
+     *
+     * @return <tt>true</tt> if this file can be efficiently moved or <tt>false</tt> otherwise
+     */
+    public boolean canFastMove() {
+        try {
+            if (canMoveHandler != null) {
+                return canMoveHandler.test(this);
+            }
+
+            return moveHandler != null;
+        } catch (Exception e) {
+            throw handleErrorInCallback(e, "canMoveHandler");
+        }
+    }
+
+    /**
+     * Tries to efficiently move the file in the given directory.
+     *
+     * @param newParent the new parent directory
+     * @return <tt>true</tt> if the operation was successful, <tt>false</tt> otherwise
+     */
+    public boolean tryFastMoveTo(VirtualFile newParent) {
+        try {
+            if (moveHandler == null) {
+                return false;
```

Comment:
I think inverting this if check would improve readability/flow and woudl be consistent to the other methods in this class